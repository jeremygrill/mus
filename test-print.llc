; ModuleID = 'Mus'

%chord_node = type { i32, %chord_node* }
%seq_node = type { %chord_node*, %seq_node* }

@fmt = private unnamed_addr constant [4 x i8] c"%d\0A\00"
@fmt.1 = private unnamed_addr constant [12 x i8] c"(%d,%d|%d)\0A\00"

declare i32 @printf(i8*, ...)

declare i32 @printc(%chord_node*)

declare i32 @prints(%seq_node*)

declare i32 @playn(i32)

declare i32 @playc(%chord_node*)

declare i32 @plays(%seq_node*)

define i32 @main() {
entry:
  %i = alloca i32
  %n = alloca i32
  %c = alloca %chord_node*
  %s = alloca %seq_node*
  store i32 555810849, i32* %n
  %c1 = alloca %chord_node*
  %malloccall = tail call i8* @malloc(i32 ptrtoint (%chord_node* getelementptr (%chord_node, %chord_node* null, i32 1) to i32))
  %a2 = bitcast i8* %malloccall to %chord_node*
  store %chord_node* %a2, %chord_node** %c1
  %a3 = load %chord_node*, %chord_node** %c1
  %a4 = getelementptr inbounds %chord_node, %chord_node* %a3, i32 0, i32 0
  store i32 926351415, i32* %a4
  %a5 = load %chord_node*, %chord_node** %c1
  %a6 = getelementptr inbounds %chord_node, %chord_node* %a5, i32 0, i32 1
  store %chord_node* null, %chord_node** %a6
  %a7 = load %chord_node*, %chord_node** %c1
  %c11 = alloca %chord_node*
  %malloccall.2 = tail call i8* @malloc(i32 ptrtoint (%chord_node* getelementptr (%chord_node, %chord_node* null, i32 1) to i32))
  %a23 = bitcast i8* %malloccall.2 to %chord_node*
  store %chord_node* %a23, %chord_node** %c11
  %a34 = load %chord_node*, %chord_node** %c11
  %a45 = getelementptr inbounds %chord_node, %chord_node* %a34, i32 0, i32 0
  store i32 1111621698, i32* %a45
  %a56 = load %chord_node*, %chord_node** %c11
  %a67 = getelementptr inbounds %chord_node, %chord_node* %a56, i32 0, i32 1
  store %chord_node* %a7, %chord_node** %a67
  %a78 = load %chord_node*, %chord_node** %c11
  store %chord_node* %a78, %chord_node** %c
  %c19 = alloca %seq_node*
  %malloccall.10 = tail call i8* @malloc(i32 trunc (i64 mul nuw (i64 ptrtoint (i1** getelementptr (i1*, i1** null, i32 1) to i64), i64 2) to i32))
  %a211 = bitcast i8* %malloccall.10 to %seq_node*
  store %seq_node* %a211, %seq_node** %c19
  %a312 = load %seq_node*, %seq_node** %c19
  %a413 = getelementptr inbounds %seq_node, %seq_node* %a312, i32 0, i32 0
  %c114 = alloca %chord_node*
  %malloccall.15 = tail call i8* @malloc(i32 ptrtoint (%chord_node* getelementptr (%chord_node, %chord_node* null, i32 1) to i32))
  %a216 = bitcast i8* %malloccall.15 to %chord_node*
  store %chord_node* %a216, %chord_node** %c114
  %a317 = load %chord_node*, %chord_node** %c114
  %a418 = getelementptr inbounds %chord_node, %chord_node* %a317, i32 0, i32 0
  store i32 1482162264, i32* %a418
  %a519 = load %chord_node*, %chord_node** %c114
  %a620 = getelementptr inbounds %chord_node, %chord_node* %a519, i32 0, i32 1
  store %chord_node* null, %chord_node** %a620
  %a721 = load %chord_node*, %chord_node** %c114
  %c122 = alloca %chord_node*
  %malloccall.23 = tail call i8* @malloc(i32 ptrtoint (%chord_node* getelementptr (%chord_node, %chord_node* null, i32 1) to i32))
  %a224 = bitcast i8* %malloccall.23 to %chord_node*
  store %chord_node* %a224, %chord_node** %c122
  %a325 = load %chord_node*, %chord_node** %c122
  %a426 = getelementptr inbounds %chord_node, %chord_node* %a325, i32 0, i32 0
  store i32 1667432547, i32* %a426
  %a527 = load %chord_node*, %chord_node** %c122
  %a628 = getelementptr inbounds %chord_node, %chord_node* %a527, i32 0, i32 1
  store %chord_node* %a721, %chord_node** %a628
  %a729 = load %chord_node*, %chord_node** %c122
  store %chord_node* %a729, %chord_node** %a413
  %a530 = load %seq_node*, %seq_node** %c19
  %a631 = getelementptr inbounds %seq_node, %seq_node* %a530, i32 0, i32 1
  store %seq_node* null, %seq_node** %a631
  %a732 = load %seq_node*, %seq_node** %c19
  store %seq_node* %a732, %seq_node** %s
  store i32 0, i32* %i
  %i33 = load i32, i32* %i
  %printf = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt, i32 0, i32 0), i32 %i33)
  %printf34 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt, i32 0, i32 0), i32 2)
  %n35 = load i32, i32* %n
  %tmp = sdiv i32 %n35, 16777216
  %tmp36 = and i32 %n35, 16777215
  %tmp37 = sdiv i32 %tmp36, 65536
  %tmp38 = and i32 %n35, 65535
  %printn = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([12 x i8], [12 x i8]* @fmt.1, i32 0, i32 0), i32 %tmp, i32 %tmp37, i32 %tmp38)
  %printn39 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([12 x i8], [12 x i8]* @fmt.1, i32 0, i32 0), i32 44, i32 44, i32 44)
  %c40 = load %chord_node*, %chord_node** %c
  %printc = call i32 @printc(%chord_node* %c40)
  %c141 = alloca %chord_node*
  %malloccall.42 = tail call i8* @malloc(i32 ptrtoint (%chord_node* getelementptr (%chord_node, %chord_node* null, i32 1) to i32))
  %a243 = bitcast i8* %malloccall.42 to %chord_node*
  store %chord_node* %a243, %chord_node** %c141
  %a344 = load %chord_node*, %chord_node** %c141
  %a445 = getelementptr inbounds %chord_node, %chord_node* %a344, i32 0, i32 0
  store i32 1296891981, i32* %a445
  %a546 = load %chord_node*, %chord_node** %c141
  %a647 = getelementptr inbounds %chord_node, %chord_node* %a546, i32 0, i32 1
  store %chord_node* null, %chord_node** %a647
  %a748 = load %chord_node*, %chord_node** %c141
  %printc49 = call i32 @printc(%chord_node* %a748)
  %s50 = load %seq_node*, %seq_node** %s
  %prints = call i32 @prints(%seq_node* %s50)
  ret i32 0
}

declare noalias i8* @malloc(i32)
