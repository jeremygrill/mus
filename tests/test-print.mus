Int main()
{
	Int i;
	Note n;
	Chord c;
	Seq s;

	n = (33,33|33);
	c = [(55,55|55),(66,66|66)];
	s = $([(88,88|88),(99,99|99)]);

	i = 0;


	print(i);
	print(2);
	printn(n);
	printn((44,44|44));
	printc(c);
	printc([(77,77|77)]);
	prints(s);

	return 0;
}
