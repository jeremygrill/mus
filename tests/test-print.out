0
2
(33,33|33)
(44,44|44)
[(55,55|55)(66,66|66)]
[(77,77|77)]
$([(99,99|99)(88,88|88)])
